pkg/services/serviceaccounts/api/api.go

```diff
@@ -132,14 +136,42 @@ func (api *ServiceAccountsAPI) ConvertToServiceAccount(ctx *models.ReqContext) r
 	}
 }
 
-func (api *ServiceAccountsAPI) ListServiceAccounts(ctx *models.ReqContext) response.Response {
-	serviceAccounts, err := api.store.ListServiceAccounts(ctx.Req.Context(), ctx.OrgId, -1)
+func (api *ServiceAccountsAPI) ListServiceAccounts(c *models.ReqContext) response.Response {
+	serviceAccounts, err := api.store.ListServiceAccounts(c.Req.Context(), c.OrgId, -1)
 	if err != nil {
 		return response.Error(http.StatusInternalServerError, "Failed to list service accounts", err)
 	}
+
+	saIDs := map[string]bool{}
+	for i := range serviceAccounts {
+		serviceAccounts[i].AvatarUrl = dtos.GetGravatarUrlWithDefault("", serviceAccounts[i].Name)
+		saIDs[strconv.FormatInt(serviceAccounts[i].Id, 10)] = true
+	}
+
+	metadata, err := api.getAccessControlMetadata(c, saIDs)
+	if err == nil && len(metadata) != 0 {
+		for i := range serviceAccounts {
+			serviceAccounts[i].AccessControl = metadata[strconv.FormatInt(serviceAccounts[i].Id, 10)]
+		}
+	}
+
 	return response.JSON(http.StatusOK, serviceAccounts)
 }
 
+func (api *ServiceAccountsAPI) getAccessControlMetadata(c *models.ReqContext, saIDs map[string]bool) (map[string]accesscontrol.Metadata, error) {
+	if api.accesscontrol.IsDisabled() || !c.QueryBool("accesscontrol") {
+		return nil, nil
+	}
+
+	userPermissions, err := api.accesscontrol.GetUserPermissions(c.Req.Context(), c.SignedInUser)
+	if err != nil || len(userPermissions) == 0 {
+		api.log.Warn("could not fetch accesscontrol metadata for teams", "error", err)
+		return nil, err
+	}
+
+	return accesscontrol.GetResourcesMetadata(c.Req.Context(), userPermissions, "serviceaccounts", saIDs), nil
```

Comment:
I was under the impression that this is a user can "create" , "edit" or "delete" a specific "scoped" resource.

But this is maybe something different